### PR TITLE
Parameter Validation requirements

### PR DESCRIPTION
**Pull request by adrianhall**

### CODE REVIEW

File: docs/design/ParameterValidation.mdk [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+The client library will have a client object with several methods that call methods on the service.  _Service parameters_ are directly passed across the wire to an Azure service.  _Client parameters_ are not passed directly to the service, but used within the SDK to fulfill the request.  Examples of client parameters include values that are used to construct a URI, or a file that needs to be uploaded to storage.
+
+~ Must {#general-parameter-validation-client} 
+validate client parameters.
+~
+
+~ MustNot {#general-parameter-validation-service}
+validate service parameters.  Let the service validate any request parameters.
+~
```
Comment (JonathanGiles) [inline docs/design/ParameterValidation.mdk]: I am fine with this advice, and will pass it on to the Java team (as we do do these checks at present). I wonder if it is worth being very explicit here and stating that this includes not doing null checks, checks for empty strings, etc as well?

File: docs/design/ParameterValidation.mdk [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+~ Must {#general-parameter-validation-errors}
+validate the developer experience when the service parameters are invalid to ensure appropriate error messages are generated by the service.
```
Comment (JonathanGiles) [inline docs/design/ParameterValidation.mdk]: I feel like this requirement doesn't clearly set out the expectation that we are placing on client library developers. How do they validate the UX? Is the expectation that the client library developer create unit tests or sample apps that pass in illegal service parameters, and to ensure that the service returns useful feedback? What should they do if it does not - does that get communicated to the architecture board or to the service team directly?

File: docs/design/ParameterValidation.mdk (hunk 1 above)
Comment (adrianhall) [reply docs/design/ParameterValidation.mdk]: I'll add something that clarifies.

File: docs/design/ParameterValidation.mdk (hunk 2 above)
Comment (adrianhall) [reply docs/design/ParameterValidation.mdk]: Good point.  I'll put some additional work into wording on this one.

**Pull request merged**